10up/distributor
# For non-super-admins, use get_blogs_of_user() to determine authroized sites.

### Description of the Change

Modifies the query used to determine a user's authorized sites to use `get_blogs_of_user()`.

As super admins have access to all sites without being a member, the existing `get_sites()` query is used for super admins.

Closes #1176

### How to test the Change


1. Install Distributor on a Multisite Install in sub-directory mode
2. Make it big `export XDEBUG_MODE=off; for i in {1..1050}; do wp site create --slug=$i --title="MS $i" --url=https://ms-distributor.local; done;` (change the URL if needs be)
3. Create a non super-admin user
4. Add the new user as an administrator on sites 2, 3 and 1050
5. Log in/switch to the new user
6. Visit the pull screen of either site 2 or 3.
7. The last site should be displayed (1050)

Creating the sites may take some time so you should leave the command going and switch tasks for a while.

### Changelog Entry

> Fixed - Improved database query for getting a users authorized sites on very large networks.


### Credits

Props @jeffpaul, @boonebgorges, @peterwilsoncc.


### Checklist:
